### DISCUSSION TITLE
How to disable warning: Line Length [Expected: 80; Actual: *]

### DISCUSSION THREAD

**Discussion by sukantamaikap** [LazyVim/LazyVim#2367] [Q&A] [answered]
I'm using LazyVim with a mostly out-of-the-box setup. I'm getting `Line Length [Expected: 80; Actual: *]` error for *.md files only. if I rename the file to *.txt these warnings go away!!

My `:LspInfo`:

```txt
 Language client log: /Users/sm/.local/state/nvim/lsp.log
 Detected filetype:   markdown
 
 4 client(s) attached to this buffer: 
 
 Client: null-ls (id: 1, bufnr: [3])
 	filetypes:       markdown, markdown.mdx, terraform, tf, terraform-vars, less, scss, css, typescript, vue, yaml, graphql, html, javascript, json, jsonc, handlebars, typescriptreact, javascriptreact, dockerfile, fish, lua, luau, sh
```

<img width="1440" alt="Screenshot 2024-01-17 at 9 33 18 pm" src="https://github.com/LazyVim/LazyVim/assets/3388408/d8b91f3e-b7dd-4dca-baca-aef1f3477511">

I disabled MD013 warning following [this](https://github.com/LazyVim/LazyVim/discussions/2268).

**Comment by dpetka2001** [comment]
This must be coming from using `null-ls` source for linting. That other answer is for `nvim-lint` used for linting. They are different. `nvim-lint` gives a different warning message `MD013/line-length Line length [Expected: 80; Actual: 116]`.

**Reply by dpetka2001** [reply]
You could disable `none-ls` all together or you could just specifically disable `markdownlint` as a `none-ls` source, so you don't get duplicate diagnostics. You could also just append the same rule as `extra_args` to the `none-ls` source, but that would be like keeping tabs on 2 different things that essentially offer the same thing. 

My personal suggestion would be to disable `none-ls` overall, but it's your decision to make depending on your workflow and what you're used to.

**Comment by sukantamaikap** [comment]
I disabled `null-is` by adding `disabled.lua` under plugins; here is the proof:
<img width="1332" alt="Screenshot 2024-01-18 at 9 14 26 pm" src="https://github.com/LazyVim/LazyVim/assets/3388408/e27cc453-0abe-45b0-b7b2-630ca03fa3f8">

I still see the same warning:

<img width="997" alt="Screenshot 2024-01-18 at 9 14 51 pm" src="https://github.com/LazyVim/LazyVim/assets/3388408/69bedbd5-13d0-4e9e-b828-de2d31e62aa1">

**Reply by dpetka2001** [reply]
LazyVim uses `none-ls` as a LazyExtra. That's where you have to disable it. Did you enable it from the Extras or did you install it yourself? From your [lazy-lock.json](https://github.com/sukantamaikap/my.lazyvim/blob/07dc66a4a9bfbc6a2827f21ee4e5e29c006fb96b/lazy-lock.json#L41) it seems that you have `none-ls.nvim` installed. That's what you have to remove.

**Reply by dpetka2001** [reply, marked as answer]
In your [lazy.lua](https://github.com/sukantamaikap/my.lazyvim/blob/07dc66a4a9bfbc6a2827f21ee4e5e29c006fb96b/lua/config/lazy.lua#L26) file you're importing the `none-ls` Extra. Remove that line. No need for adding it to your `disabled.lua`. Also you can use `:LazyExtras` command to enable/disable the LazyVim extras instead of declaring them as imports in your `lazy.lua` (from the Dashboard you can also press `x` to bring up the UI for Extras), but that is up to you what to use. Just make sure you use one of them, so there won't be any conflicts.

**Reply by sukantamaikap** [reply]
You are correct; that worked. Thanks a lot!! I added it to `lazy.lua` following a tutorial.

**Reply by tf42src** [reply]
Same issue here. No need to warn about 80 characters in modern times. Is disabling all of none-ls the right solution here? Isn't there a config to only disable `MD013` ?

**Reply by tf42src** [reply]
Adding **.markdownlint.json** to my project root with the following content worked for me:

```
{
  "MD013": false
}
```

**Reply by LorenzoBettini** [reply]
Is there a way to disable such a rule in the LazyVim configuration?

**Reply by Zhaoyilunnn** [reply]
> Is there a way to disable such a rule in the LazyVim configuration?

Same question, I really don't want to add a `.markdownlin.json` config in all my projects. Also I don't want to disable all `none-ls` diagnostics

**Reply by LorenzoBettini** [reply]
I think that's imposed by the used markdown plugin.

**Reply by Zhaoyilunnn** [reply]
> I think that's imposed by the used markdown plugin.

https://github.com/LazyVim/LazyVim/discussions/4094#discussioncomment-10178217   This solved my problem, basically add a global config and refer to it in plugin configurations

**Reply by nicc777** [reply]
> Adding **.markdownlint.json** to my project root with the following content worked for me:
> 
> ```
> {
>   "MD013": false
> }
> ```

Personally I find this to b e the most convenient solution as you can configure rules per project.

**Comment by carloshanson-itrac** [comment]
After reading the comments here, I found an alternate solution. Since LazyVim uses markdownlint-cli2, the README states that you can add a comment before the line(s) you want to disable a rule:

> Disable one or more rules by name: `<!-- markdownlint-disable MD001 MD005 -->`

I added the following to the top of my Markdown files: `<!-- markdownlint-disable MD013 -->`

https://www.lazyvim.org/extras/lang/markdown
https://github.com/DavidAnson/markdownlint
